### PR TITLE
Out of bounds warning

### PR DESCRIPTION
**Pull request by oliver-sanders**
#### tldr;
* Changed an error to a warning in cylc validate as it would drive users nuts.
* Removed timestamps from `cylc validate` log output (unless in debug mode). *Contriversial?*

#### tfm;

*Background: "Out of bounds error" refers to recurrences which lie before the `initial cycle point`.*

In #2738 (7.8.0) we added nicer handling of exceptions in `cylc validate` which can result from out of bounds recurrences in some niche cases. `cylc validate` now fails for out of bounds sequences when running in `--strict` mode for *all* cases (not just the ones resulting in traceback).

There are some cases where it might be sensible to hard code behaviour for certain cycle points up front allowing users choice of initial and final cycle points.

writing suites with out of bounds sequences might make some sort of sense, e.g. hardcoding special logic for certain dates, case studies, etc:

```ini
[scheduling]
    [[dependencies]]
        [[[P1Y]]]
            graph = foo => bar => baz
        [[[R1/2018]]]
            graph = foo => pub => baz  # 2018 was a special year
```

We see suites hardcoding cycles like this quite a lot. As `rose suite-run` invokes `cylc validate --strict` this is going to inconvenience a lot of users.

This PR demotes the error to a warning and batches up warnings if multiple out of bounds sequences are present to avoid spamming the user.

I have also removed timestamps from `cylc validate` output for readability (unless in debug mode). Time information isn't important here and the output isn't written out to log files where it might be useful.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
You have a mysterious test failure.

**Comment by oliver-sanders** [comment]
Mystery team dispatched...

File: bin/cylc-validate [hunk 1]
```diff
@@ -71,22 +74,36 @@ def main():
...
+                     if seq.get_first_point(cfg.start_point) is None]
+    if out_of_bounds:
+        if len(out_of_bounds) > 1:
+            # avoid spamming users with multiple warnings
+            msg = ('multiple sequences out of bound for initial cycle point '
+                   '%s:\n%s' % (
+                       cfg.start_point,
+                       '\n'.join(textwrap.wrap(', '.join(out_of_bounds), 70))))
+        else:
+            msg = '%s: sequence out of bound for initial cycle point %s' % (
```
Comment (hjoliver) [inline bin/cylc-validate]: Minor niggle, and it comes from the existing code, but isn't the expression "out of bound"? (plural)

**Comment by kinow** [comment]
>(I had to kick Travis to one batch of tests; looks like we have other flaky ones)

We can now add these tests to the list @matthewrmshin created: https://github.com/cylc/cylc/issues/2894 🎉

**Comment by hjoliver** [comment]
> We can now add these tests to the list @matthewrmshin created: #2894 

Unfortunately I didn't note down which ones failed, and I'm not sure if we can access eariler iterations of specific chunks of a test(?).  Will do next time!

**Comment by oliver-sanders** [comment]
> (I had to kick Travis to one batch of tests; looks like we have other flaky ones)

Looks like show/06 is quite unstable.
